docs.feldera.com/docs/pipelines/modifying.md

```diff
@@ -0,0 +1,436 @@
+This allows you to evolve your SQL pipelines by adding, removing, or modifying tables and views
+without having to re-ingest all historical data from scratch.
+
+When you modify a pipeline:
+- **New and modified views** are computed from existing pipeline state
+- **Unchanged views** continue using their existing state
+- **Input data** does not need to be re-ingested (unless table schemas change)
+
+## Introduction
+
+Feldera incrementally processes changes to input data. But what happens when the program
+itself needs to change, e.g., the user would like to add, remove, or modify a view definition?
+Feldera is able to handle such changes incrementally as well by only recomputing
+the modified view and any views derived from it. It uses existing state in tables
+and internal indexes maintained by the incremental view maintenance engine
+as input to the computation.
+
+This process of building new or modified views from existing state in the pipeline
+is called **bootstrapping**. Bootstrapping allows you to evolve your SQL pipeline without
+having to re-ingest all historical data from scratch.
+
+<!-- Unless one of the table definitions changes as well, bootstrapping new or modified
+views doesn't require re-ingesting input data. -->
+
+## How bootstrapping works
+
+In this section we describe steps involved in modifying a pipeline while preserving its state.
+
+### 1. Stop the pipeline
+
+The pipeline must be stopped before it can be modified. During the `/stop` operation, the state
+of the pipeline is checkpointed in storage. Alternatively, if the pipeline crashes or gets force-stopped,
+its recent state is preserved in the last [periodic checkpoint](/pipelines/fault-tolerance), as long as checkpointing is
+enabled.  This state will later be used to bootstrap the modified pipeline.
+
+### 2. Modify the pipeline
+
+The user modifies the pipeline by making arbitrary changes to its SQL code.
+This way many coordinated changes across multiple tables and views can be performed at
+the same time. This is particularly useful when working with nested views, where
+a change to one of the views, such as adding a new column, may require updates to all
+downstream views that depend on it.
+
+### 3. Restart the pipeline
+
+When a modified pipeline is started, it performs the following operations:
+
+1. Opens its latest checkpoint if one exists.
+
+2. Compares the checkpointed program with the new version being started.
+   If the program has not been modified since the checkpoint was made, proceeds
+   with normal initialization.
+
+3. If the program has been modified, the behavior depends on the `bootstrap_policy`
+   argument passed to the pipeline on startup:
+
+   * `bootstrap_policy=await_approval` (default) - the pipeline stops in the `AwaitingApproval`
+     state. Its `deployment_runtime_status_details` property lists tables, views, and connectors
+     affected by the changes. The user can either approve the changes by calling the
```

Comment:
at this point the pipeline is paused until the user signals their intention.
this can be done using the sdk, the webui, or using an http call.